db/db.go

```diff
@@ -0,0 +1,56 @@
+package db
+
+import (
+	"database/sql"
+
+	"github.com/pkg/errors"
+
+	_ "github.com/mattn/go-sqlite3"
+
+	"github.com/uptrace/bun"
+	"github.com/uptrace/bun/dialect/sqlitedialect"
+	"github.com/uptrace/bun/extra/bundebug"
+)
+
+var DB *RootDB
+
+type RootDB = bun.DB
+
+func NewDB(dbPath string) (RootDB, error) {
+	sqldb, err := sql.Open("sqlite3", dbPath)
+	if err != nil {
+		return RootDB{}, err
+	}
+
+	// create the database if it doesn't exist
+	// XXX this'll be in bun migrations.
+	_, err = sqldb.Exec(`
+		CREATE TABLE IF NOT EXISTS builds (
+			build_num INTEGER PRIMARY KEY,
+			url TEXT NOT NULL,
+			branch TEXT NOT NULL,
+			subject TEXT NOT NULL,
+			'commit' TEXT NOT NULL,
+			parallel INTEGER NOT NULL,
+			workflow TEXT,
+			start_time TEXT NOT NULL,
+			lifecycle TEXT NOT NULL,
+			-- this is our own metadata
+			archived BOOLEAN NOT NULL DEFAULT FALSE
+		);
+		CREATE TABLE IF NOT EXISTS artifacts (
+			url TEXT PRIMARY KEY,
+			build_num INTEGER NOT NULL
+		);
+	`)
+	if err != nil {
+		sqldb.Close()
+		return RootDB{}, errors.Wrap(err, "creating tables")
+	}
+
+	db := bun.NewDB(sqldb, sqlitedialect.New())
+	// db.AddQueryHook(bundebug.NewQueryHook(bundebug.WithVerbose(true)))
+	db.AddQueryHook(bundebug.NewQueryHook())
+	DB = db
+	return *DB, nil
```

Comment:
`*DB` strikes me as "not safe in general"... do you know that passing bun.DB by value is actually acceptable?

Reply:
this is also an omission - I did it like this just to make it compatible with the other code.